src/clr-core/internal/decorators/query-slot.ts

```diff
@@ -20,21 +20,42 @@ const standardQuery = (descriptor: PropertyDescriptor, element: any) => ({
   descriptor,
 });
 
+export interface QuerySlotConfig {
+  required?: 'error' | 'warning';
+  requiredMessage?: string;
+}
+
 /**
  * A property decorator that converts a class property into a getter that
  * executes a querySelector on the element's light DOM Slot.
  *
  * @ExportDecoratedItems
  */
-export function querySlot(selector: string) {
+export function querySlot(selector: string, config?: QuerySlotConfig) {
   return (
     protoOrDescriptor: {} | any,
     // tslint:disable-next-line:no-any decorator
     name?: PropertyKey
   ): any => {
     const descriptor = {
       get(this: LitElement) {
-        return this.querySelector(selector);
+        const ref = this.querySelector(selector);
+
+        if (config) {
+          const message = config.requiredMessage
```

Comment:
(nit) This can maybe be simplified to...

```
const message = config.requiredMessage || `The <${selector}> element is required to use <${this.tagName.toLocaleLowerCase()}>`;
```